specs/altair/sync-protocol.md

```diff
@@ -265,4 +438,473 @@ def process_light_client_update(store: LightClientStore,
+
+###### `optimistic_light_client_update`
+
+This topic is used to propagate the latest `OptimisticLightClientUpdate` to light
+clients, allowing them to stay in sync with the network.
+
+The following validations MUST pass before forwarding the `optimistic_update`
+on the network.
+- _[IGNORE]_ The optimistic update is not attesting to the latest block's parent block --
+  i.e. validate that `hash_tree_root(optimistic_update.attested_header) == state.latest_block_header.parent_root`.
+- _[REJECT]_ The optimistic update does not match the expected value --
+  i.e. keep track of the most recent `OptimisticLightClientUpdate` locally using
+  `create_optimistic_light_client_update`, and only propagate exact copies of it.
+
+To form an `OptimisticLightClientUpdate`, the following are needed:
+- `state`: latest state with a block
+- `block`: latest block (`state` is `block`'s post state)
+- `attested_state`: previous latest state with a block (`state.latest_block_header.parent_root`s post state)
+
+```python
+def create_optimistic_light_client_update(state: BeaconState,
+                                          block: SignedBeaconBlock,
+                                          attested_state: BeaconState) -> Optional[OptimisticLightClientUpdate]:
+    if not hasattr(attested_state, "next_sync_committee"):
+        return None
+
+    if sum(block.message.body.sync_aggregate.sync_committee_bits) < MIN_SYNC_COMMITTEE_PARTICIPANTS:
+        return None
+
+    block_header = BeaconBlockHeader(
+        slot=block.message.slot,
+        proposer_index=block.message.proposer_index,
+        parent_root=block.message.parent_root,
+        state_root=block.message.state_root,
+        body_root=hash_tree_root(block.message.body),
+    )
+
+    header = state.latest_block_header.copy()
+    header.state_root = hash_tree_root(state)
+    assert header == block_header
+    assert header.slot == state.slot
+    period = compute_sync_committee_period(compute_epoch_at_slot(header.slot))
+
+    attested_header = attested_state.latest_block_header.copy()
+    attested_header.state_root = hash_tree_root(attested_state)
+    assert hash_tree_root(attested_header) == state.latest_block_header.parent_root
+    assert attested_header.slot == attested_state.slot
+    attested_period = compute_sync_committee_period(compute_epoch_at_slot(attested_header.slot))
+
+    if period not in (attested_period, attested_period + 1):
+        return None
+
+    return OptimisticLightClientUpdate(
+        attested_header=attested_header,
+        sync_aggregate=block.message.body.sync_aggregate,
+        fork_version=state.fork.current_version,
+        is_signed_by_next_sync_committee=(period == attested_period + 1),
+    )
+```
+
+### The Req/Resp domain
+
+#### Messages
+
+##### BestLightClientUpdatesByRange
+
+**Protocol ID:** `/eth2/beacon_chain/req/best_light_client_updates_by_range/1/`
+
+Request Content:
+```
+(
+  start_period: uint64
+  count: uint64
+  step: uint64
```

Comment:
Can we drop the `step` parameter? For lightclient updates it doesn't make sense since you must process strictly sequential updates.

Reply:
Well, the client could still use multiple parallel downloads with non-1 steps to try and get a faster speed. It is very similar to `beaconBlocksByRange` which, ultimately, also need to be applied in sequence. If a client is not interested in using that optimization, just supply `1` as step size. Personally I'd prefer leaving it in for consistency with `beaconBlocksByRange`, also enabling code reuse with whatever downloader is being used for blocks.

Reply:
Note that PR https://github.com/ethereum/consensus-specs/pull/2850 for EIP-4844 is dropping the step parameter intentionally breaking the beacon_blocks_by_range pattern

https://github.com/ethereum/consensus-specs/pull/2850/files#diff-c055ad660df5d383113f454c99ab8d6f70964643405d019ca43560ec9ac2841bR181-R182

Reply:
Oh, that's interesting, given the trivial implementation on the server side and the optionality on the client side. The referenced PR refers to "previous complaints dev-feedback" in the description – is there any additional background on these? If this is a general trend to no longer use `step` for newer endpoints (besides just PR specific style), sure I can also remove it here for consistency going forward.

Reply:
More background on "previous complaints dev-feedback": https://github.com/ethereum/consensus-specs/pull/2856